thorlcr/graph/thgraphslave.hpp

```diff
@@ -65,11 +65,13 @@ class CEdgeProgress
 #ifdef _TESTING
         owner.ActPrintLog("ITDL starting for output %d", outputId);
 #endif
-#ifdef _TESTING
-        bool started = hasStarted();
-        bool stopped = hasStopped();
-        assertex(!hasStarted() || hasStopped());      // ITDL started twice
-#endif
+        if (hasStarted())
+        {
+            if (!hasStopped())
+                throw MakeActivityException(&owner, 0, "Starting without being stopped 1st");
+            else
+                throw MakeActivityException(&owner, 0, "Started and stopped states both set");
+        }
```

Comment:
Not directly related to the bug, but would have made the error hit/being debugged easier to diagnose.

Reply:
I agree the new code is clearer.   Does this change anything (other than improving the error message?)  I don't think started and stopped can ever be set at the same time.  (If so how can they?)

Reply:
it doesn't change anything, purely to make error clearer.
I didn't think they could both be set at the same time, but was reluctant to remove that check in case could (e.g. corrupted)